src/entity/Unit.ts

```diff
@@ -1350,47 +1356,49 @@ export function inRange(unit: IUnit, target: Vec2): boolean {
 }
 
 export async function startTurnForUnits(units: IUnit[], underworld: Underworld, prediction: boolean) {
-  // Trigger start turn events
-  const turnStartPromises = [];
-  for (let unit of units) {
-    turnStartPromises.push(runTurnStartEvents(unit, underworld, prediction))
+  // Add mana to Player units
+  for (let unit of units.filter(u => u.unitType == UnitType.PLAYER_CONTROLLED && u.alive)) {
+    // Restore player to max mana at start of turn
+    // Let mana remain above max if it already is
+    // (due to other influences like mana potions, spells, perks, etc);
+    unit.mana = Math.max(unit.manaMax, unit.mana);
   }
-  await raceTimeout(5000, 'Turn Start Events did not resolve', Promise.all(turnStartPromises));
```

Comment:
What's the reasoning for the changes in this file?

Reply:
More descriptive comments would be helpful.  This comment just says "Fix Turn Start/End organization " but that provides no context about why the change was made and what's improved.  Descriptive commit messages are especially helpful when looking back in time to see why something was changed long ago.

Reply:
The initial problem was that dark tide and freeze were setting stamina to 0 in turnStart events, but units were immediately regenerating stamina *after* the turn start events finished. I thought about the ordering more, and the way I had written it previously didn't make as much sense, it was a mistake when I did the turn phase refactor. Afaik the only real effect of the ordering was undoing freeze's stamina change, but this didnt cause any issues in-game since freeze now uses canAct and that sort've overrides whether or not a unit can do things, so stamina at that point doesnt matter.

This was just cleanup for all of that it makes sense that units would generate stamina first (or players generate mana first), before we run a bunch of turn start logic that might then affect those stats.

Reply:
That makes sense! Thanks!